experimental/rsocket-src/RSocketConnectionHandler.cpp

```diff
@@ -45,29 +45,29 @@ class RSocketHandlerBridge : public reactivesocket::DefaultRequestHandler {
         std::move(streamId));
     // bridge from the existing eager RequestHandler and old Subscriber type
     // to the lazy Flowable and new Subscriber type
-    flowable->subscribe(yarpl::Reference<yarpl::flowable::Subscriber<Payload>>(
-        new NewToOldSubscriber(std::move(response))));
+    flowable->subscribe(std::move(response));
 
     return eagerSubscriber;
   }
 
   void handleRequestResponse(
       Payload request,
       StreamId streamId,
-      const std::shared_ptr<Subscriber<Payload>>&
+      const yarpl::Reference<yarpl::flowable::Subscriber<Payload>>&
```

Comment:
We can change this to take SingleObserver<Payload> instead of FlowableSubscriber and then eliminate the bridge. But that may need to be a followup change as I imagine that has far reaching impact lower down.